src/net.rs

```diff
@@ -673,7 +673,7 @@ unsafe fn connect_overlapped(
 
     let ptr = CONNECTEX.get(socket)?;
     assert!(ptr != 0);
-    let connect_ex = mem::transmute::<_, LPFN_CONNECTEX>(ptr);
+    let connect_ex = mem::transmute::<_, LPFN_CONNECTEX>(ptr).unwrap();
```

Comment:
In each case, the assert on the previous line already validates that the `Option` callback can be unwrapped. Perhaps there's a simpler way to write this or the assert is now just redundant.